include/grpcpp/ext/gcp_observability.h

```diff
@@ -17,25 +17,76 @@
 #ifndef GRPCPP_EXT_GCP_OBSERVABILITY_H
 #define GRPCPP_EXT_GCP_OBSERVABILITY_H
 
+#include <grpc/support/port_platform.h>
+
 #include "absl/status/status.h"
+#include "absl/status/statusor.h"
 
 namespace grpc {
-namespace experimental {
+
+namespace grpc_gcp {
+
+// Observability objects follow the RAII idiom and help manage the lifetime of
+// gRPC Observability data exporting to GCP. It is illegal to directly construct
+// objects of this class. `ObservabilityInit()` should be invoked instead to
+// return an `Observability` instance. Observability data is flushed at regular
+// intervals, and also when this instance goes out of scope and its destructor
+// is invoked.
+class Observability {
+ public:
+  ~Observability();
+
+  // Move constructor. The moved-from object will no longer be valid and will
+  // not cause GCP Observability stats, tracing and logging data to flush.
+  Observability(Observability&& other) noexcept;
+
+  // Delete copy and copy-assignment operator
+  Observability(const Observability&) = delete;
+  Observability& operator=(const Observability&) = delete;
+
+ private:
+  friend absl::StatusOr<Observability> ObservabilityInit();
+  Observability() = default;
+  bool close_on_destruction_ = true;
+};
 
 // Initialize GCP Observability for gRPC.
 // This should be called before any other gRPC operations like creating a
 // channel, server, credentials etc.
 // The most common usage would call this at the top (or near the top) in main().
+// The return value helps determine whether observability was successfully
+// enabled or not. On success, an object of class `Observability` is returned.
+// When this object goes out of scope, GCP Observability stats, tracing and
+// logging data is flushed. On failure, the status and status message can be
+// used to determine the cause of failure. It is up to the applications to
+// either crash on failure, or continue without GCP observability being enabled.
+//
+// Please look at
+// https://github.com/grpc/grpc/blob/master/examples/cpp/gcp_observability/helloworld/greeter_client.cc
+// and
+// https://github.com/grpc/grpc/blob/master/examples/cpp/gcp_observability/helloworld/greeter_server.cc
+// for sample usage.
+//
+// Note that this is a blocking call which properly sets up gRPC Observability
+// to work with GCP and might take a few seconds to return.  Similarly, the
+// destruction of a non-moved-from `Observability` object is also blocking since
+// it flushes the observability data to GCP.
+//
 // As an implementation detail, this properly initializes the OpenCensus stats
 // and tracing plugin, so applications do not need to perform any additional
 // gRPC C++ OpenCensus setup/registration to get GCP Observability for gRPC.
-absl::Status GcpObservabilityInit();
+absl::StatusOr<Observability> ObservabilityInit();
```

Comment:
Ok with this, but unconvinced it needs to be `absl::StatusOr` -- my expectation here is that `absl::optional` would be just find for real usage, and mean that folks don't need to interpret status codes.

If we keep `StatusOr` here, I'd encourage some documentation around how to interpret the status codes.

Reply:
I have the same opinion as Mark when preferring absl::StatusOr<> over absl::optional<> here because of the human-readable error message.
I do think though that in the future, we *might* be able to provide some explicit guidance based on the status code, for example, retry if unvailable. For now, though, I'll add that users should not make any assumptions about the behavior based on the resulting status code.